esdt-safe/src/to_sovereign/create_tx.rs

```diff
@@ -112,114 +115,205 @@ pub trait CreateTxModule:
         self.remove_items(&mut self.banned_endpoint_names(), &all_names);
     }
 
-    /// Create an Elrond -> Sovereign transaction.
     #[payable("*")]
-    #[endpoint]
-    fn deposit(
-        &self,
-        to: ManagedAddress,
-        opt_transfer_data: OptionalValue<TransferData<Self::Api>>,
-    ) {
+    #[endpoint(depositBack)]
+    fn deposit_back(&self, to: ManagedAddress) {
         require!(self.not_paused(), "Cannot create transaction while paused");
 
+        let caller = self.blockchain().get_caller();
+        require!(caller == ESDT_SYSTEM_SC_ADDRESS.into(), "Caller is invalid");
+
+        let payments = self.call_value().all_esdt_transfers();
```

Comment:
probably would be a good idea to keep the checks for payment not being empty and the number of tokens not exceeding maximum.

Reply:
done